assignments/pull_requests.go

```diff
@@ -0,0 +1,157 @@
+package assignments
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/autograde/quickfeed/database"
+	"github.com/autograde/quickfeed/kit/score"
+	"github.com/autograde/quickfeed/scm"
+)
+
+var (
+	// These are used to track how many times someone has been assigned to review a pull request. They map as follows.
+	// teacherReviewCounter[courseID][userID] = count
+	// groupReviewCounter[groupID][userID] = count
+	teacherReviewCounter = make(map[uint64]map[uint64]int)
+	groupReviewCounter   = make(map[uint64]map[uint64]int)
+)
+
+// CreateFeedbackComment formats a feedback comment to be posted on pull requests.
+// It uses the test results from a student commit to create a table like the one shown below.
+// Only the test scores associated with the supplied task are used to generate this table.
+// Table formatting ref: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables
+//
+//  ## Test results from latest push
+//	| Test Name | Score | Weight | % of Total |
+//	| :-------- | :---- | :----- | ---------: |
+//  | Test 1	| 2/6	| 1		 |	   12.23% |
+//  | Test 2	| 1/3   | 2	     |     24.46% |
+//  |   |		|  |    | |      |            |
+//  |   |		|  |    | |      |            |
+//  |   |		|  |    | |      |            |
+//  |   |		|  |    | |      |            |
+//  | Total		|		|		 |	   48.23% |
+// Once a total score of 80% is reached, reviewers are automatically assigned.
+//
+func CreateFeedbackComment(results *score.Results, task *pb.Task, assignment *pb.Assignment) string {
+	body := "## Test results from latest push\n"
+	body += "| Test Name | Score | Weight | % of Total |\n"
+	body += "| :-------- | :---- | :----- | ---------: |\n"
+
+	for _, score := range results.Scores {
+		if score.TaskName != task.LocalName() {
+			continue
+		}
+		percentageScore := (float64(score.Score) / float64(score.MaxScore)) * (float64(score.Weight) / results.TotalTaskWeight(task.LocalName()))
```

Comment:
We already calculate something similar in `kit/score/results.go`. This computation should also be done in this file.